src/System.Windows.Extensions/tests/System/Media/SoundPlayerTests.cs

```diff
@@ -0,0 +1,421 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+using System.IO;
+using System.Runtime.Serialization.Formatters.Binary;
+using Xunit;
+
+namespace System.Media.Test
+{
+    public class SoundPlayerTests
+    {
+        [Fact]
+        public void Ctor_Default()
+        {
+            var player = new SoundPlayer();
+            Assert.Null(player.Container);
+            Assert.False(player.IsLoadCompleted);
+            Assert.Equal(10000, player.LoadTimeout);
+            Assert.Null(player.Site);
+            Assert.Empty(player.SoundLocation);
+            Assert.Null(player.Stream);
+            Assert.Null(player.Tag);
+        }
+
+        public static IEnumerable<object[]> Stream_TestData()
+        {
+            yield return new object[] { null };
+            yield return new object[] { new MemoryStream() };
+        }
+
+        [Theory]
+        [MemberData(nameof(Stream_TestData))]
+        public void Ctor_Stream(Stream stream)
+        {
+            var player = new SoundPlayer(stream);
+            Assert.Null(player.Container);
+            Assert.False(player.IsLoadCompleted);
+            Assert.Equal(10000, player.LoadTimeout);
+            Assert.Null(player.Site);
+            Assert.Empty(player.SoundLocation);
+            Assert.Same(stream, player.Stream);
+            Assert.Null(player.Tag);
+        }
+
+        [Theory]
+        [InlineData("http://google.com")]
+        [InlineData("invalid")]
+        [InlineData("/file")]
+        public void Ctor_String(string soundLocation)
+        {
+            var player = new SoundPlayer(soundLocation);
+            Assert.Null(player.Container);
+            Assert.False(player.IsLoadCompleted);
+            Assert.Equal(10000, player.LoadTimeout);
+            Assert.Null(player.Site);
+            Assert.Equal(soundLocation ?? "", player.SoundLocation);
+            Assert.Null(player.Stream);
+            Assert.Null(player.Tag);
+        }
+
+        [Theory]
+        [InlineData(null)]
+        [InlineData("")]
+        public void Ctor_NullOrEmptyString_ThrowsArgumentException(string soundLocation)
+        {
+            AssertExtensions.Throws<ArgumentException>("path", null, () => new SoundPlayer(soundLocation));
+        }
+
+        public static IEnumerable<object[]> Play_String_TestData()
+        {
+            yield return new object[] { "C:/Users/hughb/Downloads/adpcm.wav" };
+            yield return new object[] { "C:/Users/hughb/Downloads/pcm.wav" };
```

Comment:
@ViktorHofer @safern do we have replacement for these test data?

Reply:
You need to use the package created in the corefx-testdata repo: https://github.com/dotnet/corefx-testdata/tree/master/System.Windows.Extensions.TestData (this is the package: https://dotnet.myget.org/feed/dotnet-core/package/nuget/System.Windows.Extensions.TestData)

In order to use it you need to add a reference to it in the XuniRuntime.depproj file, something like: https://github.com/dotnet/corefx/pull/33092/files#diff-841e9e59d27acec708f2c73eaa3cf6a0R80

Once you add that reference now you need to copy over the contents of the package to the test woriking directory in your test's csproj, something like: https://github.com/dotnet/corefx/pull/33092/files#diff-4472d7978ec00068bd255d07a778935bR160

Once you've done that, you can only refer here to the relative path of the files within the test working directory.

Reply:
Excellent description of what's needs to be done. Thanks Santi :)